### PR TITLE
Remote step executor

### PR DESCRIPTION
**Pull request by iv-p**
## Description

Added remote step executor and a new command for the CLI - to execute steps remotely, on apid infrastructure.

### Type of change

- New feature (non-breaking change which adds functionality)

### Checklist:

- [ x ] I have performed a self-review of my own code
- [ x ] Go code is formatted with `gofmt`
- [ x ] I have made corresponding changes to the docs in `/docs` and in the `README`
- [ x ] I have added tests (unit and/or end-to-end) that prove my fix is effective or that my feature works
- [ x ] Any dependent changes have been merged and published (in downstream modules)

### CODE REVIEW

File: common/step/remote_executor.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+	req.Header.Set("Authorization", e.key)
+	res, err := e.c.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode == native.StatusTooManyRequests {
+		log.L.Fatal("resource quota exceeded")
```
Comment (dimitarvdimitrov) [inline common/step/remote_executor.go:56]: return an error, the request runner (common/step/remote_executor.go) handles errors gracefully already

File: svc/cli/cmd/remote_check.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	apid remote -c ./e2e-tests/ -k <apid access key> -r us-east`,
+	Args: cobra.NoArgs,
+	RunE: remoteRun,
+}
+
+func init() {
+	rootCmd.AddCommand(remoteCmd)
+	remoteCmd.Flags().StringVarP(&configFilepath, "config", "c", "./apid.yaml", "file with config to run")
+	remoteCmd.Flags().BoolVarP(&showTimings, "timings", "t", false, "output the durations of requests")
+	remoteCmd.Flags().StringVarP(&apiKey, "key", "k", "", "apid access key")
```
Comment (dimitarvdimitrov) [inline svc/cli/cmd/remote_check.go:44]: we should (also) get this from the environment (amend docs when changed)

File: common/step/remote_executor.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	"bytes"
+	"io/ioutil"
+	native "net/http"
+	"os"
+
+	"github.com/getapid/apid-cli/common/log"
+
+	"github.com/getapid/apid-cli/common/http"
+	"gopkg.in/yaml.v2"
+)
```
[the remark this answers is not in the document]
Comment (iv-p) [reply common/step/remote_executor.go]: Done, for some reason my autoformatter didn't work

File: common/step/remote_executor.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	c        *native.Client
+}
+
+// NewRemoteHTTPExecutor instantiates a new http executor
+func NewRemoteHTTPExecutor(client *native.Client, apiKey string, region string) Executor {
+	endpointProvider := endpoint.NewAPIDEndpointProvider()
+	endpoint, err := endpointProvider.GetForRegion(region)
+	if err != nil {
+		log.L.Error(err)
+		os.Exit(1)
```
[the remark this answers is not in the document]
Comment (iv-p) [reply common/step/remote_executor.go]: Changed.

File: common/step/remote_executor.go (hunk 1 above)
Comment (iv-p) [reply common/step/remote_executor.go]: Done.

File: svc/cli/cmd/remote_check.go (hunk 2 above)
Comment (iv-p) [reply svc/cli/cmd/remote_check.go]: Changed code and docs.

**Pull request merged**